### ISSUE TITLE
Please add release version numbers tags

### ISSUE THREAD

**Issue by bmiller59** [spantaleev/matrix-docker-ansible-deploy#762]
Because this project is changing so quickly and because updates are so valuable, it would be wonderful to be able to "watch releases" on Github to be notified when it was time to pull and deploy new changes.

Can you add version tags when you push updates?

Thank for you this great project!

**Comment by StanESmith** [comment]
Agreed, this would be a valuable addition. Checking the CHANGELOG every once and a while is good but having the addition of release version numbers and the tags would benefit everyone.

**Comment by bmiller59** [comment]
Using semver would also give the project the flexibility to introduce breaking changes, if necessary, on major version numbers, etc.

**Comment by spantaleev** [comment]
Even now, we introduce breaking changes all the time. We have to, in order to keep going forward.

Major releases obviously imply minor releases, so it's a complication -- us having to maintain multiple branches and to make even more releases as a whole. It encourages people to stay behind (for how long?) and that has the chance to introduce even more problems once they ultimately decide to make a big jump in time. Some people do that even now and it's a support trouble. Once we make it "the norm", it might be worse.

How minor should releases be? Synapse moves pretty fast and has the potential to break things. Do we update Synapse in a minor release or hedge that risk and only update other (smaller things)? Is there a point to such releases though. I guess not. We need to upgrade Synapse. And although I haven't really encountered troubles with Synapse upgrades, it's possible, especially if you rely on integrations (matrix-corporal had become broken after Synapse upgrades; so have other components - password providers, etc). Which components should be upgraded in a minor release and which shouldn't? Or maybe we only update components, but don't do other "more breaking" changes (like migrating all your bridges from SQLite to Postgres?)?

I think it's too much work and it encourages a bad habit. Matrix (and Synapse, etc.) still move fast, so there's some breakage sometimes and there's the need to update often. Even now we have trouble with people not performing some upgrade tasks (Postgres upgrades, especially around the `C` collation change, announced in the changelog), and then tripping up with other updates 1 year later. We probably have people who are still using SQLite for their bridges, even now we default everyone to Postgres (unless you opt-out).

**Comment by bmiller59** [comment]
I think it might actually help people to stay up to date and would hopefully not require more effort on your part. 

Watching all changes on this repo results in a firehose of alerts, which are too much for almost everyone except maintainers. But if you are not watching those changes, you do not know when important things have been updated and may delay upgrades simply because one is not aware that changes have been made.

I personally would be very happy with a very simple versioning system like this:
* Use major versions for breaking changes that require manual intervention
* Use patch versions for everything else
* (Optional) Use minor version when completely new functionality has been added that is backward compatible

I don't think you need to add new branches necessarily. 

I am grateful you maintain this repo so well, so the simplest possible versioning that does not require significant changes to your workflow would already be appreciated.

I would like to unwatch all changes on this repo and watch only for releases. That will help me to upgrade more regularly/automatically, and alert me when there are breaking changes where I should read the changelog.

Thank you.

**Comment by aaronraimist** [comment]
Imo it is unnecessary for the purpose of knowing when to update. If you want a regular reminder to update you can follow the releases of the Synapse GitHub repo. There is a new Synapse version every ~2 weeks so if you update every time Synapse gets a new release then you are staying pretty up to date.

If the decision is made to have matrix-docker-ansible-deploy publish releases then I think it would not make sense to use semver. Instead a release could be published each time there is a breaking change mentioned in `CHANGELOG.md`. The version number would just be the current date. Then you could follow MDAD releases to get a notification of breaking changes in the playbook and Synapse (or another component) to get notifications of other regular updates.

**Comment by enekonieto** [comment]
I think bmiller just asks to introduce a version number with informational purposes, only as a brief of the changelog. For example, I could see the version change and wonder:
* This is only a patch, I'm not updating, with my internet connection running the ansible takes me almost an hour (real).
* A new minor version, interesting, I'll read the changelog for the new functions.
* A major version, I need to carefully study the breaking changes!

**Comment by rltas** [comment]
Not sure I can follow that logic when the whole playbook relies on every default component it pulls to set proper tags. If I don't update my standalone Synapse for a year I'll face the same problems, with or without the playbook; similarly, once people got their stack up, at least part of them will probably not pull again, with or without tags.

**Comment by mark-monteiro** [comment]
For anyone just looking to get notified when there are changes in `CHANGELOG.md`, I found a couple quick ways to do this from [this](https://stackoverflow.com/q/19169787/1988326) StackOverflow answer:

- https://app.github-file-watcher.com/ is a website that lets you monitor changes to specific files on GitHub. FYI it's not an open source project in case that matters to you
- GitHub offers RSS feeds for individual files. The one in question you would need in this case would be this: https://github.com/spantaleev/matrix-docker-ansible-deploy/commits/master/CHANGELOG.md.atom

**Comment by spantaleev** [comment]
Closing this one, as the decision here has effectively been made and has held for years: the playbook intentionally does not publish releases or version tags.

As discussed above, this project tracks fast-moving upstream components (Synapse, bridges, etc.), so breaking changes are a regular and necessary part of moving forward. Cutting releases would add maintenance overhead and would encourage staying on old versions, which tends to make eventual upgrades harder, not easier. The supported workflow remains: track the `master` branch, check [CHANGELOG.md](https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/master/CHANGELOG.md) for backward-incompatible changes, then pull and re-run the playbook (see [Upgrading the Matrix services](https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/master/docs/maintenance-upgrading-services.md)).

For the notification use case that motivated this request, you can subscribe to the changelog directly via GitHub's per-file feed, without watching the whole repository: https://github.com/spantaleev/matrix-docker-ansible-deploy/commits/master/CHANGELOG.md.atom

Thank you all for the thoughtful discussion. If circumstances change significantly in the future, this can certainly be revisited.
